xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxBXA.h

```diff
@@ -56,17 +56,19 @@ class CDVDDemuxBXA : public CDVDDemux
 
   bool Open(CDVDInputStream* pInput);
   void Dispose();
-  void Reset();
-  void Abort();
-  void Flush();
-  DemuxPacket* Read();
+
+  // implementation of CDVDDemux
+  virtual void Reset() override;
+  virtual void Abort() override;
+  virtual void Flush() override;
+  virtual DemuxPacket* Read() override;
   bool SeekTime(double time, bool backwards = false, double* startpts = NULL) override { return false; }
-  void SetSpeed(int iSpeed) {};
-  int GetStreamLength() { return (int)m_header.durationMs; }
+  virtual void SetSpeed(int iSpeed) override {}
```

Comment:
demuxer could handle this internally. it knows about the nature of the stream and if it should be paused. if read is not called for a given time, it can pause ffmpeg. when player continues reading, the stream can continue as well. why should upper layers bother with those internals.